BUILD.bazel

```diff
@@ -0,0 +1,166 @@
+# Allow any ts_library rules in this workspace to reference the config
+exports_files(["tsconfig.json"], visibility = ["//:__subpackages__"])
+
+package(default_visibility = ["//visibility:public"])
+
+load("@build_bazel_rules_nodejs//:index.bzl", "nodejs_binary", "nodejs_test")
+load("@npm_bazel_typescript//:index.bzl", "ts_project", "ts_config")
+load("@npm_bazel_labs//:index.bzl", "protobufjs_ts_library")
+
+proto_library(
+    name = "google_protobuf_protos",
+    srcs = glob(["node_modules/google-gax/build/protos/google/protobuf/*.proto"]) +
+    [
+        "node_modules/google-gax/build/protos/google/protobuf/compiler/plugin.proto",
+    ],
+    strip_import_prefix = "node_modules/google-gax/build/protos",
+)
+
+proto_library(
+    name = "google_rpc_protos",
+    deps = [
+        ":google_protobuf_protos",
+    ],
+    srcs = glob(["node_modules/google-gax/build/protos/google/rpc/*.proto"]),
+    strip_import_prefix = "node_modules/google-gax/build/protos/",
+)
+
+proto_library(
+    name = "google_api_protos",
+    deps = [
+        ":google_protobuf_protos",
+    ],
+    srcs = glob(["node_modules/google-gax/build/protos/google/api/*.proto"]),
+    strip_import_prefix = "node_modules/google-gax/build/protos",
+)
+
+proto_library(
+    name = "google_longrunning_protos",
+    deps = [
+        ":google_protobuf_protos",
+        ":google_api_protos",
+        ":google_rpc_protos",
+    ],
+    srcs = ["node_modules/google-gax/build/protos/google/longrunning/operations.proto"],
+    strip_import_prefix = "node_modules/google-gax/build/protos",
+)
+
+proto_library(
+    name = "service_config_proto",
+    deps = [
+        ":google_protobuf_protos",
+        ":google_rpc_protos",
+    ],
+    srcs = ["protos/service_config.proto"],
+    strip_import_prefix = "protos/",
+)
+
+protobufjs_ts_library(
+    name = "protos",
+    deps = [
+        ":google_api_protos",
+        ":google_longrunning_protos",
+        ":service_config_proto",
+    ]
+)
+
+filegroup(
+    name = "templates",
+    srcs = glob(["templates/**/*.*", "templates/**/.*"]),
+)
+
+filegroup(
+    name = "test-fixtures",
+    srcs = glob(["test-fixtures/**/*.*"]),
+)
+
+filegroup(
+    name = "baselines",
+    srcs = glob(["baselines/**/*.*", "baselines/**/.*"]),
+)
+
+npm_runtime_dependencies = [
+    "@npm//file-system",
```

Comment:
How does this interact with the dependencies defined in `package.json`?  Do you need to maintain this list in two places?

Reply:
Yes. https://bazelbuild.github.io/rules_nodejs/Built-ins.html
![image](https://user-images.githubusercontent.com/4015807/81439067-a4eafa80-9122-11ea-96ba-b245f644ce45.png)

Reply:
So ... what happens here when renovate comes along and updates our `package.json` and `package-lock.json`?

Reply:
Nothing bad - next time bazel build is run, it will figure out that `package-lock.json` has changed, and will fetch new dependencies. (I assume renovate knows how to deal with `package-lock.json`, I think it does)

Reply:
Note that here in `BUILD` file it's just a list of packages without versions, so version updates by renovate are not going to affect this in any way.

Reply:
Ahhhh - what happens when you add a new package to `package.json`?

Reply:
You go here, decide if it's a compile, or test, or runtime dependency, and add it manually to one of the lists. They call it "fine graned npm dependencies", you see.

How it works internally: after they do `yarn install` or `npm install` (whatever), they actually bazelify all the dependencies (auto-generate `BUILD.bazel` file for all of them) so that they become "targets" in bazel world (think `make` targets). Making bazel task depend on "just all the crap from `node_modules`" is not something they approve (that was deprecated long time ago), so yes, we list dependencies in two places.